### ISSUE TITLE
Using ionicons 5 in react

### ISSUE THREAD

**Issue by designerdarpan** [ionic-team/ionicons#835]
How to use it in react application , i found react-ionicon package but its kind of outdated and supports ionicons 3 , is there any way i can import ionicons 5 in my react application. 
I kind of work around with cdn and it seems to work fine , but i couldnt work around with node modules.

**Comment by dotNetkow** [comment]
We definitely need to include better docs for those, sorry! For the time being, check out our [first app tutorial](https://ionicframework.com/docs/react/your-first-app) to see how; specifically look at the [demo app](https://github.com/ionic-team/photo-gallery-capacitor-react).

**Comment by designerdarpan** [comment]
So when do you plan to include it ?

**Comment by lucej** [comment]
Yeah the official IonIcon documentation is completely wrong for a react Ionic app. 

No idea if this is official or up to date but this worked for me:

- Replace **ion-icon** with **IonIcon**
- Replace the attribute **name** with **icon**
- Wrap icon name in **brackets** and change to **camelCase**
- In the file be sure to **import** the icon from "ionicons/icons"

Instead of:
`<ion-icon name="airplane-outline"></ion-icon>`

For a react Ionic app it should be:
`<IonIcon icon={airplaneOutline}></IonIcon>`

And be sure to import the icon reference:
`import {arrowBackCircleOutline} from "ionicons/icons";`

**Comment by designerdarpan** [comment]
@lucej  this will import base64 image and should be enclosed inside a img tag , which would make changing color difficult , masking it would be an option but i need a simple way to do it. 
i did a simple hack , works a gem.
making a component and replacing base64 image files such that i would only get svg markup
```
import React from 'react'
export default function IonIconComponent(props) {
    let icon = props.icon.replace("data:image/svg+xml;utf8,","");
    return (
        <div className="ion-icon-container">
            <div className="icon-inner" dangerouslySetInnerHTML={{__html:icon}}/>
        </div>
    )
}
```
now i can use this same component wherever i need 
`
import {createOutline} from 'ionicons/icons/index';
export const ActionComponent = withRouter((props) => {
    const data = props;
  return (
    <>
          <IonIconComponent icon={createOutline}/>
    </>
  )
})
`

**Comment by designerdarpan** [comment]
@lucej this will import base64 image and should be enclosed inside a img tag , which would make changing color difficult , masking it would be an option but i need a simple way to do it.
i did a simple hack , works a gem.
making a component and replacing base64 image files such that i would only get svg markup

import React from 'react'
export default function IonIconComponent(props) {
    let icon = props.icon.replace("data:image/svg+xml;utf8,","");
    return (
        <div className="ion-icon-container">
            <div className="icon-inner" dangerouslySetInnerHTML={{__html:icon}}/>
        </div>
    )
}

now i can use this same component wherever i need

```
import {createOutline} from 'ionicons/icons/index';
export const ActionComponent = withRouter((props) => {
    const data = props;
  return (
    <>
          <IonIconComponent icon={createOutline}/>
    </>
  )
})
```

**Comment by hrghafoori** [comment]
Just use a JavaScript type in your script element. it works for me.
`<script type="JavaScript" src="https://unpkg.com/ionicons@5.2.3/dist/ionicons.js"></script>`

**Comment by wangzuo** [comment]
https://github.com/swiftcarrot/react-ionicons we transfer svg from ionicons into react component with the help of svgr

**Comment by christian-bromann** [comment]
Consolidating into #1448
